diode-server/netbox/virtualization_wrappers.go

```diff
@@ -0,0 +1,1680 @@
+
+		if !actualGroup.HasChanged() {
+			group = &VirtualizationClusterGroup{
+				ID: actualGroup.ID(),
+			}
+
+			intendedGroupID := intendedGroup.ID()
+			if intended.Cluster.Group != nil {
+				intendedGroupID = intended.Cluster.Group.ID
+			}
+
+			intended.Cluster.Group = &VirtualizationClusterGroup{
+				ID: intendedGroupID,
+			}
+		}
+
+		vw.Cluster.Group = group
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, groupObjectsToReconcile...)
+
+		tagsToMerge := mergeTags(vw.Cluster.Tags, intended.Cluster.Tags, intendedNestedObjects)
+
+		if len(tagsToMerge) > 0 {
+			vw.Cluster.Tags = tagsToMerge
+		}
+
+		for _, t := range vw.Cluster.Tags {
+			if t.ID == 0 {
+				vw.objectsToReconcile = append(vw.objectsToReconcile, &TagDataWrapper{Tag: t, hasParent: true})
+			}
+		}
+
+		actualHash, _ := hashstructure.Hash(vw.Data(), hashstructure.FormatV2, nil)
+		intendedHash, _ := hashstructure.Hash(intended.Data(), hashstructure.FormatV2, nil)
+
+		reconciliationRequired = actualHash != intendedHash
+	} else {
+		siteObjectsToReconcile, siteErr := actualSite.Patch(intendedSite, intendedNestedObjects)
+		if siteErr != nil {
+			return nil, siteErr
+		}
+
+		site, err := copyData(actualSite.Data().(*DcimSite))
+		if err != nil {
+			return nil, err
+		}
+		site.Tags = nil
+
+		if !actualSite.HasChanged() {
+			site = &DcimSite{
+				ID: actualSite.ID(),
+			}
+		}
+		vw.Cluster.Site = site
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, siteObjectsToReconcile...)
+
+		typeObjectsToReconcile, typeErr := actualType.Patch(intendedType, intendedNestedObjects)
+		if typeErr != nil {
+			return nil, typeErr
+		}
+
+		vType, err := copyData(actualType.Data().(*VirtualizationClusterType))
+		if err != nil {
+			return nil, err
+		}
+		vType.Tags = nil
+
+		if !actualType.HasChanged() {
+			vType = &VirtualizationClusterType{
+				ID: actualType.ID(),
+			}
+		}
+		vw.Cluster.Type = vType
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, typeObjectsToReconcile...)
+
+		groupObjectsToReconcile, groupErr := actualGroup.Patch(intendedGroup, intendedNestedObjects)
+		if groupErr != nil {
+			return nil, groupErr
+		}
+
+		group, err := copyData(actualGroup.Data().(*VirtualizationClusterGroup))
+		if err != nil {
+			return nil, err
+		}
+		group.Tags = nil
+
+		if !actualGroup.HasChanged() {
+			group = &VirtualizationClusterGroup{
+				ID: actualGroup.ID(),
+			}
+		}
+		vw.Cluster.Group = group
+
+		vw.objectsToReconcile = append(vw.objectsToReconcile, groupObjectsToReconcile...)
+
+		tagsToMerge := mergeTags(vw.Cluster.Tags, nil, intendedNestedObjects)
+
+		if len(tagsToMerge) > 0 {
+			vw.Cluster.Tags = tagsToMerge
+		}
+
+		for _, t := range vw.Cluster.Tags {
+			if t.ID == 0 {
+				vw.objectsToReconcile = append(vw.objectsToReconcile, &TagDataWrapper{Tag: t, hasParent: true})
+			}
+		}
+
+	}
+
+	if reconciliationRequired {
+		vw.hasChanged = true
+		vw.objectsToReconcile = append(vw.objectsToReconcile, vw)
+	}
+
```

Comment:
To ensure we don't have duplicated objects to reconcile, we should use `dedupObjectsToReconcile` func:
```suggestion

        dedupObjectsToReconcile, err := dedupObjectsToReconcile(vw.objectsToReconcile)
	if err != nil {
		return nil, err
	}
	vw.objectsToReconcile = dedupObjectsToReconcile

```